src/bpftrace.cpp

```diff
@@ -104,237 +107,111 @@ Probe BPFtrace::generate_probe(const ast::AttachPoint &ap, const ast::Probe &p)
-            auto found = target_map.find(ap.target);
-            if (found != target_map.end()) {
-              found->second.funcs.push_back(func);
-            } else {
-              auto probe = generate_probe(ap, p);
-              probe.funcs.push_back(func);
-              target_map.insert({ { ap.target, probe } });
-            }
-          }
-          for (auto &pair : target_map) {
-            resources.probes.push_back(std::move(pair.second));
-          }
-          continue;
+          auto probe = generate_probe(match_ap, p);
+          probe.funcs.push_back(func);
+          target_map.insert({ { match_ap.target, probe } });
         }
       }
-    } else if ((probetype(attach_point->provider) == ProbeType::uprobe ||
-                probetype(attach_point->provider) == ProbeType::uretprobe ||
-                probetype(attach_point->provider) == ProbeType::watchpoint ||
-                probetype(attach_point->provider) ==
-                    ProbeType::asyncwatchpoint) &&
-               !attach_point->func.empty()) {
-      std::set<std::string> matches;
-
-      struct symbol sym = {};
-      int err = resolve_uname(attach_point->func, &sym, attach_point->target);
-
-      if (attach_point->lang == "cpp") {
-        // As the C++ language supports function overload, a given function name
-        // (without parameters) could have multiple matches even when no
-        // wildcards are used.
-        matches = probe_matcher_->get_matches_for_ap(*attach_point);
+      for (auto &pair : target_map) {
+        resources.probes.push_back(std::move(pair.second));
       }
-
-      if (err >= 0 && sym.address != 0)
-        matches.insert(attach_point->target + ":" + attach_point->func);
-
-      attach_funcs.insert(attach_funcs.end(), matches.begin(), matches.end());
     } else {
-      if (probetype(attach_point->provider) == ProbeType::usdt &&
-          !attach_point->ns.empty())
-        attach_funcs.push_back(attach_point->target + ":" + attach_point->ns +
-                               ":" + attach_point->func);
-      else if (probetype(attach_point->provider) == ProbeType::tracepoint ||
-               probetype(attach_point->provider) == ProbeType::uprobe ||
-               probetype(attach_point->provider) == ProbeType::uretprobe ||
-               probetype(attach_point->provider) == ProbeType::kfunc ||
-               probetype(attach_point->provider) == ProbeType::kretfunc)
-        attach_funcs.push_back(attach_point->target + ":" + attach_point->func);
-      else if ((probetype(attach_point->provider) == ProbeType::kprobe ||
-                probetype(attach_point->provider) == ProbeType::kretprobe) &&
-               !attach_point->target.empty()) {
-        attach_funcs.push_back(attach_point->target + ":" + attach_point->func);
-      } else {
-        attach_funcs.push_back(attach_point->func);
-      }
+      std::vector<std::string> funcs(matches.begin(), matches.end());
+      probe.funcs = funcs;
```

Comment:
```suggestion
      probe.funcs(matches.begin(), matches.end());
```
In-place construction to avoid copying the vector and strings

Reply:
I used `probe.funcs.assign(matches.begin(), matches.end())` which is more compact and (IIUC) should have the same complexity.